File: ios/Sources/Mia21/Models/TranscriptionModels.swift
//
//  TranscriptionModels.swift
//  Mia21
//
//  Created on November 4, 2025.
//  Copyright © 2025 Mia21. All rights reserved.
//
//  Description:
//  Speech-to-text transcription models.
//  Includes transcription response with text, language, and duration.
//

import Foundation

// MARK: - Transcription Response

public struct TranscriptionResponse: Codable {
  public let text: String
  public let language: String?
  public let duration: Double?
  
  // Note: CodingKeys not needed - all properties match JSON keys exactly
}

File: android/lib/src/main/kotlin/com/mia21/models/TranscriptionModels.kt
/**
 * TranscriptionModels.kt
 * Mia21
 *
 * Created on November 25, 2025.
 * Copyright © 2025 Mia21. All rights reserved.
 *
 * Description:
 * Speech-to-text transcription models.
 * Includes request and response models for audio transcription.
 */

package com.mia21.models

import kotlinx.serialization.Serializable

/**
 * Response from audio transcription
 */
@Serializable
data class TranscriptionResponse(
    val text: String,
    val language: String? = null,
    val confidence: Double? = null
)

